cmd/dex/serve.go

```diff
@@ -508,19 +513,53 @@ func runServe(options serveOptions) error {
 		}
 
 		grpcSrv := grpc.NewServer(grpcOptions...)
+		healthcheck := health.NewServer()
+		healthgrpc.RegisterHealthServer(grpcSrv, healthcheck)
 		api.RegisterDexServer(grpcSrv, server.NewAPI(serverConfig.Storage, logger, version, serv))
 
 		grpcMetrics.InitializeMetrics(grpcSrv)
 		if c.GRPC.Reflection {
 			logger.Info("enabling reflection in grpc service")
 			reflection.Register(grpcSrv)
 		}
+		ctx, cancelHealthcheck := context.WithCancel(context.Background())
```

Comment:
do you need a ctx here? 
I think `group` does the trick with canceling the function.

Can you check it?

Reply:
I need context for early breaking from the `for { }` loop below when shutdown is requested.

Indeed, `group` cancels the function, but after the `gracefulStop` of grpc server times out (see my comment below). Meanwhile, the healhcheck loop keeps updating status if not properly stopped from the outside (which is not a big deal, but I prefer stopping stuff when it is not needed anymore).

I can use bare channel for receiving shutdown event, but context is more convenient.